src/modules/Game.class.js

```diff
@@ -1,68 +1,364 @@
-'use strict';
-
-/**
- * This class represents the game.
- * Now it has a basic structure, that is needed for testing.
- * Feel free to add more props and methods if needed.
- */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.initialState = initialState.map((row) => [...row]);
+    this.state = initialState;
+    this.score = 0;
+    this.status = 'idle';
+    this.messages = [...document.querySelectorAll('.message')];
+  }
+
+  moveLeft() {
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    for (let i = 0; i < this.state.length; i++) {
+      let row = this.state[i];
+
+      row = row.filter((val) => val !== 0);
+
+      for (let j = 0; j < row.length - 1; j++) {
+        if (row[j] === row[j + 1]) {
+          row[j] *= 2;
+          this.score += row[j];
+          row[j + 1] = 0;
+          this.moved = true;
+        }
+      }
+
+      row = row.filter((val) => val !== 0);
+
+      while (row.length < 4) {
+        row.push(0);
+      }
+
+      if (row.toString() !== this.state[i].toString()) {
+        this.moved = true;
+      }
+
+      this.state[i] = row;
+    }
+
+    if (this.moved) {
+      this.addRandomTile();
+      this.getScore();
+      this.updateDOM();
+      this.getStatus();
+    }
+  }
+
+  moveRight() {
+    if (this.status !== 'playing') {
+      return;
+    }
+
+    for (let i = 0; i < this.state.length; i++) {
+      let row = this.state[i];
+
+      row = row.filter((val) => val !== 0);
+      row.reverse();
+
+      for (let j = 0; j < row.length - 1; j++) {
+        if (row[j] === row[j + 1]) {
+          row[j] *= 2;
+          this.score += row[j];
+          row[j + 1] = 0;
+          this.moved = true;
+        }
+      }
```

Comment:
This part (almost all method actually) is exactly the same as in moveLeft method

Please, try to reuse some logic between "move" functions